### DISCUSSION TITLE
Typed constants are not constants?

### DISCUSSION THREAD

**Discussion by alfaus** [vlang/v#19570] [Questions and Answers] [answered]
If you use a typed constant in a match range expression, for example:
```
const (
  a = u16(1)
  b = u16(5)
)
c := u16(3)
match c {
  a...b { println('1...5') }
  else { println('not 1...5') }
}
```
you get a compiler error: 'match branch range expressions need the start value to be known at compile time (only enums, const or literals are supported)'
Maybe u16(1) is not a type conversion but a function call? I've read functions in constant definition are evaluated at runtime.

I don't know if this is a bug or a feature...

**Comment by Delta456** [comment]
This should work.

**Reply by Delta456** [reply, marked as answer]
Should be fixed in https://github.com/vlang/v/pull/19572

**Reply by Delta456** [reply]
The PR is merged!

**Reply by alfaus** [reply]
Thanks!
